tcl/util.tcl

```diff
@@ -542,3 +542,42 @@ proc qc::not_null {var} {
     upvar $var value
     return [expr {[info exists value] && $value ne "" }]
 }
+
+proc qc::glob_recursive {args} {
+    #| Return names of files that match patterns, including files in sub-directories
+    # Usage: same as glob (see http://www.tcl.tk/man/tcl8.5/TclCmd/glob.htm),
+    # with additional max_depth switch
+    qc::args $args -join -nocomplain -tails -path ? -directory ? -types ? -max_depth 10 -- args
+
+    # Use a single glob call, but extend each pattern passed in with */ prefixes
+    # (eg. *.html becomes *.html */*.html */*/*.html etc.)
+    if { [info exists join] } {
```

Comment:
I included -join because `glob` has it, and I figured anyone using `glob_recursive` would expect it to have the same options as `glob` unless there was a good reason not to.

That said, I can't figure out why the developers of tcl felt the need to include this option in the first place, so I guess I'd be equally happy to take it out altogether.

(Incidentally, `glob -join path1/folder/ path2/folder` is equivalent to `glob path1/folder//path2/folder`. I don't know if that makes sense, but it is how tcl does it).